tests/Microsoft.Bot.Builder.Prompts.Tests/ChoicePromptTests.cs

```diff
@@ -0,0 +1,169 @@
+                        "blue"
+                    },
+                    {
+                        new Regex("alizarin crimson|amaranth|american rose|auburn|blood red|burgundy|cardinal|carmine|carnelian|cerise|chocolate cosmos|cinnabar|coquelicot|crimson|dark red|electric crimson|fire brick|flame|folly|fuchsia|indian red|magenta|maroon|mahogany|mystic red|oxblood|persian red|pink|raspberry|red|red-violet|redwood|rose|rosewood|rosso corsa|ruby|rust|scarlet|terra cotta|turkey red|tuscan red|tyrian purple|venetian red|vermilion|wine", RegexOptions.IgnoreCase | RegexOptions.Singleline),
+                        "red"
+                    }
+                };
+            }
+
+            public IDictionary<Regex, string> MapRegexes { get; private set; }
+            public Regex TokenRegex { get; private set; }
+            public bool AllowPartialMatch => false;
+            public int MaxDistance => 2;
+            public bool OnlyTopMatch => true;
+        }
+
+        internal class ColorParserConfiguration : IChoiceParserConfiguration<Color>
+        {
+            public ColorParserConfiguration()
+            {
+                Resolutions = new Dictionary<string, Color>()
+                {
+                    { "red", Color.Red },
+                    { "blue", Color.Blue },
+                    { "yellow", Color.Yellow }
+                };
+            }
+
+            public IDictionary<string, Color> Resolutions { get; private set; }
+
+        }
+
+        internal class ColorModel : ChoiceModel
+        {
+            public override string ModelTypeName => "colour";
+
+            public ColorModel() : base(
+                new OptionsParser<Color>(new ColorParserConfiguration()),
+                new ChoiceExtractor(new ColorExtractorConfiguration()))
+            {
+            }
+
+            protected override SortedDictionary<string, object> GetResolution(ParseResult parseResult)
+            {
+                var data = parseResult.Data as OptionsParseDataResult;
+                return new SortedDictionary<string, object>()
+                {
+                    { "value", parseResult.Value },
+                    { "score", data.Score },
+                    { "otherResults", data.OtherMatches.Select(l => new { l.Text, l.Value, l.Score })},
+                };
+            }
+        }
+    }
+}
```

Comment:
Wouldn't ['yellow', 'blue', 'red'] be a simpler way to express a list of choices?  Even expanding to include a list of synonyms shouldn't be more complicated then passing in an array of `Choice` objects.